src/api/entities/Identity/index.ts

```diff
@@ -667,9 +681,18 @@ export class Identity extends Entity<UniqueIdentifiers, string> {
       },
     } = this;
 
-    const assembleResult = ({ secondary_keys: secondaryKeys }: DidRecord): SecondaryKey[] => {
+    const assembleResult = ({
+      secondary_keys: secondaryKeys,
+    }: DidRecord): PermissionedAccount[] => {
       return secondaryKeys.map(({ signer: rawSigner, permissions }) => ({
-        signer: signerValueToSigner(signatoryToSignerValue(rawSigner), context),
+        account: new Account(
+          {
+            address: signerToString(
```

Comment:
This looks like a lot of nested helper function calls. I think a new function `signerToAddress` could make this nicer.

Reply:
Actually, since Identities can no longer be secondary keys, we can assume the rawSigner here always represents an Account

This means we can do one of the following:

- Use a type assertion `account: signerValueToSigner(signatoryToSignerValue(rawSigner), context) as Account`
- Create a `signatoryToAccount` conversion util and use it here

In both cases we should add a comment explaining that even though the chain types support identities as secondary accounts, in practice it doesn't occur and it will be removed

```
/**
 * @hidden
 */
export function signatoryToAccount(signatory: Signatory, context: Context): Account {
  if (signatory.isAccount) {
    return new Account({ address: accountIdToString(signatory.asAccount) }, context);
  }

  throw new PolymeshError({ code: ErrorCode.UnexpectedError, message: 'Received an Identity where an Account was expected. Please report this issue to the Polymath team' });
}
```

I suggest the second approach. More explicit and robust

Reply:
Added the new util method